docs/Roadmap.md

```diff
@@ -36,35 +25,75 @@ instance. Care must be taken to keep the standalone version usable.
+we can legally do so. That also means that URLs should live forever once they are created, which places a burden on us
+to keep existing compilers and libraries available forever.
 
 ### Non-goals
 
 Compiler Explorer will remain open-source and non-commercial. There are no plans at all to add "freemium" content. We do
-have a Patreon site, GitHub sponsors, PayPal donations, and some corporate sponsors. Funds from these source help
-support the cost of running the servers, and incentivize the core team. Previously our goal was to remain "ad-free",
-that has been relaxed slightly to allow up to three company sponsor logos visible at the top right of the screen.
+have a Patreon site, GitHub sponsors, PayPal donations, and some corporate sponsors, for which we are incredibly
+grateful. Funds from these source help support the cost of running the servers, save and plan for the future, and
+incentivize the core team. Previously our goal was to remain "ad-free", that has been relaxed slightly to allow up to
+three company sponsor logos visible at the top right of the screen.
 
 ## Goals
 
-### 2021 goals
-
-- **Login support**. Support logging in to the site with GitHub, Google, etc. We will _never_ force you to log in for
-  basic features, and of course will update the Privacy Policy. I won't be selling anything to do with user info etc,
-  either: Logging in will be purely used to make _your_ life easier and allow you to manage things like shared settings
-  and configuration, listing short URLs you've created (and potentially being able to remove them); and _maybe_ being
-  able to make user-named short URLs (e.g., "godbolt.org/u/mattgodbolt/ctad-example"). This goal is a personal pet
-  project of [Matt's](http://github.com/mattgodbolt/).
-- **Multi-file support**. Multiple file compilation units to open the door to seeing LTO and maybe modules. This may
-  include being able to use a `CMake` file to build things.
-- **Modernising the codebase**. Moving the codebase to TypeScript, or something similar that will allow us to worry less
-  about differences between front-end and back-end code (old JavaScript versions), and help us attract more people to
-  the project.
-- **More compilers and libraries**. Plus finishing off the last stragglers of installation.
+### 2023 goals
+
+- **Modernising the codebase**. 2023 has to be the year of all Typescript. We need to continue pushing towards this goal
+  as we are currently blocked using old versions of some libraries due to JS/TS/node.js/ES6 modules conflicts. We're
+  still using an outdated and unsupported module loader and in order to move off it we need to update to a newer
+  loader...that itself doesn't seemingly doesn't support a mixture of typescript and javascript like we currently have.
+  In any case, moving to Typescript has improved our code quality and continues to find bugs, so we should absolutely
+  port to it for this reason alone.
```

Comment:
Excellent, I think we should definitely look to get this done!

- JS -> TS changes have been excellent this year, and I'm so excited to see that we're now 60% TS on the GitHub graph!
- We've had a few attempts and chats about ESM, but it seems we're definitely moving - hoping to be able to land on something after the TS conversion like we've discussed

Reply:
I'm hoping so: yes. It's becoming harder and harder to do anything without hitting ESM problems.